src/EnergyPlus/extendedHI.cc

```diff
@@ -0,0 +1,369 @@
+                        tol,
+                        maxIter,
+                        SolFla,
+                        Ts,
+                        [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (phi_salt * pvstar(Ts) - Pa) / Za_un - (Q - Qv(Ta, Pa)); },
+                        0.0,
+                        Tc);
+                    Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                    eqvar_name = "Rs*";
+                }
+            } else {
+                Rs = 0.0;
+                eqvar_name = "dTcdt";
+                dTcdt = (1.0 / C) * flux3;
+            }
+        }
+        return {eqvar_name, phi, Rf, Rs, dTcdt};
+    }
+
+    // Convert the find_T function
+    std::pair<Real64, std::string> find_T(EnergyPlusData &state, std::string eqvar_name, Real64 eqvar)
+    {
+        Real64 T;
+        std::string region;
+        int SolFla;
+
+        if (eqvar_name == "phi") {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<1>(find_eqvar(state, T, 1.0)) - eqvar; }, 0.0, 240.0);
+            region = "I";
+        } else if (eqvar_name == "Rf") {
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                T,
+                [&](Real64 T) { return std::get<2>(find_eqvar(state, T, std::min(1.0, Pa0 / pvstar(T)))) - eqvar; },
+                230.0,
+                300.0);
+            region = (Pa0 > pvstar(T)) ? "II" : "III";
+        } else if (eqvar_name == "Rs" || eqvar_name == "Rs*") {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<3>(find_eqvar(state, T, Pa0 / pvstar(T))) - eqvar; }, 295.0, 350.0);
+            region = (eqvar_name == "Rs") ? "IV" : "V";
+        } else {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return std::get<4>(find_eqvar(state, T, Pa0 / pvstar(T))) - eqvar; }, 340.0, 1000.0);
+            region = "VI";
+        }
+
+        return std::make_pair(T, region);
+    }
+
+    Real64 heatindex(EnergyPlusData &state, Real64 Ta, Real64 RH, bool show_info = false)
+    {
+
+        // Ta: temperature in Kelvin
+        // RH: relative humidity in range of 0.0 to 1.0
+        // show_info: whether to print some messages. Might be useful in debugging
+        // The function computes the extended heat index, in Kelvinn
+
+        auto const HVACSystemRootSolverBackup = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
+        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverAlgorithm::Bisection;
+        // Dictionary to map eqvar_name to tuple index
+        std::map<std::string, int> dic = {{"phi", 1}, {"Rf", 2}, {"Rs", 3}, {"Rs*", 3}, {"dTcdt", 4}};
+        auto eqvars = find_eqvar(state, Ta, RH);
+        std::string eqvar_name = std::get<0>(eqvars);
```

Comment:
Are you asking a function to return a string, and then doing a string comparison to decide which code path to follow?  For one thing, this arrangement seems way way overkill just to calculate heat index.  And second, it cannot be a string, you need to _at least_ use an enum.  But I really feel this whole thing could be way simpler.

Reply:
I will simplify this